src/cpp/utils/threading.hpp

```diff
@@ -52,25 +69,43 @@ void set_name_to_current_thread(
 /**
  * @brief Give a name to the thread calling this function.
  *
- * @param[in]  fmt   A null-terminated string to be used as the format argument of
- *                   a `snprintf` like function, in order to accomodate the restrictions of
- *                   the OS. Those restrictions may truncate the final thread name if there
- *                   is a limit on the length of the name of a thread.
- * @param[in]  arg1  First variadic argument passed to the formatting function.
- * @param[in]  arg2  Second variadic argument passed to the formatting function.
+ * @param[in, out]  thread_name_buffer  Buffer to store the name of the thread.
+ * @param[in]       fmt   A null-terminated string to be used as the format argument of
+ *                        a `snprintf` like function, in order to accomodate the restrictions of
+ *                        the OS. Those restrictions may truncate the final thread name if there
+ *                        is a limit on the length of the name of a thread.
+ * @param[in]       arg1  First variadic argument passed to the formatting function.
+ * @param[in]       arg2  Second variadic argument passed to the formatting function.
  */
 void set_name_to_current_thread(
+        char* thread_name_buffer,
         const char* fmt,
         uint32_t arg1,
         uint32_t arg2);
 
+/**
+ * @brief Give a name to the thread calling this function.
+ *
+ * @param[in]       fmt   A null-terminated string to be used as the format argument of
+ *                        a `snprintf` like function, in order to accomodate the restrictions of
+ *                        the OS. Those restrictions may truncate the final thread name if there
+ *                        is a limit on the length of the name of a thread.
+ * @param[in]       arg1  First variadic argument passed to the formatting function.
+ * @param[in]       arg2  Second variadic argument passed to the formatting function.
+ */
+void set_name_to_current_thread(
```

Comment:
Why is this one necessary?
Are we calling this outside `eprosima::create_thread`?